src/System.IO.Pipelines/tests/SchedulerFacts.cs

```diff
@@ -291,11 +397,8 @@ private void Work(object state)
 
             async Task DoRead()
             {
-                int oid = Thread.CurrentThread.ManagedThreadId;
```

Comment:
Why removed?

Reply:
The thread could be reused. I saw it fail once in a local run.

Reply:
This way test tests nothing, if test starts on threadpool thread and scheduler just runs inline it won't fail

Reply:
What about this, I'll only assert that thread isn't a thread pool thread to begin with, if it is we'll know why it's failing if it does fail randomly. I'll also add a check to make sure it doesn't complete synchronously.

Reply:
> I'll assert that thread isn't a thread pool thread to begin with

Then keep the other asserts too, they wouldn't fail in other case.

Reply:
Got it! Confirmed that we do start on the thread pool sometimes. I'll make a manual thread as you suggest.

Reply:
@pakrym Yea I added them all back and added more as well.